www-apps/radicale/radicale-2.1.8.ebuild

```diff
@@ -0,0 +1,76 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="6"
+
+PYTHON_COMPAT=( python{3_4,3_5,3_6} )
+
+inherit eutils distutils-r1 user
+
+MY_PN="Radicale"
+MY_P="${MY_PN}-${PV}"
+
+DESCRIPTION="A simple CalDAV calendar server"
+HOMEPAGE="http://radicale.org/"
+SRC_URI="mirror://pypi/${MY_PN:0:1}/${MY_PN}/${MY_P}.tar.gz"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="+bcrypt"
+
+RDEPEND=">=dev-python/vobject-0.9.5[${PYTHON_USEDEP}]
+	bcrypt? ( dev-python/passlib[bcrypt,${PYTHON_USEDEP}] )"
+
+S=${WORKDIR}/${MY_P}
+
+RDIR=/var/lib/radicale
+LDIR=/var/log/radicale
+
+pkg_pretend() {
+	if [[ -f ${RDIR}/.props && ${MERGE_TYPE} != buildonly ]]; then
+		eerror "It looks like you have a version 1 database in ${RDIR}."
+		eerror "You must convert this database to version 2 format before upgrading."
+		eerror "You may want to back up the old database before migrating."
+		eerror "Please see the migration guide for details:"
+		eerror "  http://radicale.org/1to2/"
+		eerror "After conversion, please delete the old database (including hidden .props files),"
+		eerror "then copy the exported database into ${RDIR}."
+		eerror "You should finish with a ${RDIR}/collection-root directory."
```

Comment:
Something to ponder upon, also in relevance to the news item: the upstream 'migration guide' is quite poor IMHO. The export stuff is basically mentioned in middle of paragraph without even a proper example, so I would suggest giving users some explicit suggestion, preferably something they could copy&paste. E.g.:

    some-command --export-storage /var/lib/radicale.new
    mv /var/lib/radicale /var/lib/radicale.old
    mv /var/lib/radicale.new /var/lib/radicale

This sample covers conversion, and also backup. You can suggest upgrading, testing and removing the backup if it works.

Reply:
I guess the migration instructions would be best placed in the news item, not the ebuild, right? Because that’s where most people will see them first, and the ebuild can just tell people to go read the news item? I agree the guide is not all that helpful for Gentoo users, but then I think it’s written more for people who have customized their install paths and everything. If you agree that this should go in the news item, then I’ll post a new revision with more detailed instructions in it.

As an aside, would you be willing to sponsor my continued posting access to the mailing list? I don’t know when the posting restrictions will happen, but if they do happen before this all gets sorted out, then I would not be able to post the new revisions of the news item. Alternatively I could send the news item somewhere (to you?) to pass on to the list.

Reply:
I'd personally go for both, but up to you. As for ml, don't worry about it. When infra starts doing something, I'll probably look into including all active proxy-maint members.

Reply:
How does 7817dd8f3cc53a6fae4a11180695865297b0f874 look? If you think the instructions are fine, then I will convert them to news item format and, barring any further issues, squash the branch.